app/master/subjob.py

```diff
@@ -103,29 +98,6 @@ def atomic_commands(self):
         job_command = self.job_config.command
         return ['{} {}'.format(atom.command_string, job_command) for atom in self._atoms]
 
-    def get_atom_artifact_name(self, atom_id):
```

Comment:
E.g. the following `artifact_dir` method it has the information about `build_id`, 'subjob_id'. Given an instance of a `Subjob`, it's just easier to get the artifact_dir by calling a method with `atom_id` than calling a static method with `build_id`, etc. The code can stay dry as you can call the static method as the implementation of `artifact_dir` here. This might not be a good example. But the main thing I want to share is that if the object already has the information, asking the consumer of the object to fetch the property from the object (`build_id`, etc.) and then call a static method is a little more work than just using the internal properties and call a method on the object instead.